driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -172,6 +172,16 @@ public static ArrayExpression<StringExpression> ofStringArray(final String... ar
         return new MqlExpression<>((cr) -> new AstPlaceholder(new BsonArray(result)));
     }
 
+    public static <R extends DocumentExpression> R current() {
```

Comment:
While we don't know how will `Expression.isDocumentOr` return values of user-supplied subtypes of `DocumentExpression`, or whether it will even be possible to implement the idea of user subtypes, it at least it does not immediately seem absolutely impossible.

Returning a value of a user-defined type from a `static` method `Expressions.current()` is, on the other hand, absolutely impossible, provided that we exclude any kind of global state meddling (which we should). Thus, `Expressions.current` should return `DocumentExpression`.